#! /bin/sh /usr/share/dpatch/dpatch-run
## 05_segfault_params.dpatch by  <Jonathan Wiltshire <debian@jwiltshire.org.uk>>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't segfault on arbitrary parameters

@DPATCH@
diff -urNad gxemul-0.4.6.6~/src/main.c gxemul-0.4.6.6/src/main.c
--- gxemul-0.4.6.6~/src/main.c	2008-11-16 15:35:57.000000000 +0000
+++ gxemul-0.4.6.6/src/main.c	2008-11-24 11:08:05.000000000 +0000
@@ -714,6 +714,13 @@
 
 	get_cmd_args(argc, argv, emul, &diskimages, &n_diskimages);
 
+	if (emul->machines[0]->machine_type == MACHINE_NONE) {
+		printf("No machine type specified? Run  gxemul -H  for a list\n"
+		    "of available machine types. Use the -e or -E option(s)\n"
+		    "to specify the machine type.\n");
+		exit(1);
+	}
+
 	if (!skip_srandom_call) {
 		struct timeval tv;
 		gettimeofday(&tv, NULL);
